src/components/AppointmentForm.tsx

```diff
@@ -0,0 +1,173 @@
+'use client'
+
+import { useTransition } from 'react'
+import { useForm } from 'react-hook-form'
+import { appointmentFormAction } from './actions'
+import type AppointmentFormData from '@/types/AppointmentFormData'
+import styles from '@styles/appointmentForm.module.css'
+
+const AppointmentForm = () => {
+  const [, startTransition] = useTransition()
+
+  const defaultValues = {
+    email: '',
+    dob: '',
+    apptDate: '',
+    apptType: '',
+    optionalDescription: '',
+    emergency: '',
+  }
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isValid, isDirty },
+  } = useForm<AppointmentFormData>({
+    defaultValues,
+    mode: 'onChange',
+  })
+
+  const onSubmit = handleSubmit((data) => {
+    console.log(data)
+    startTransition(() => {
+      appointmentFormAction(data)
+    })
+  })
+
+  return (
+    <form
+      className={styles.apptForm}
+      onSubmit={(event) => void onSubmit(event)}
+    >
+      <div className={styles.apptFormSection}>
+        <label htmlFor="email" className={styles.inputLabel}>
+          Email:{' '}
+        </label>
+        <input
+          type="text"
+          id="email"
+          className={styles.input}
+          placeholder="Enter email"
+          {...register('email', {
+            required: 'Email is required.',
+            validate: {
+              maxLength: (v) =>
+                v.length <= 50 || 'The email should have at most 50 characters',
+              matchPattern: (v) =>
+                /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/.test(v) ||
+                'Email address must be a valid address',
+            },
+          })}
+        />
+        <p className={styles.error}>{errors.email && errors.email.message}</p>
+      </div>
+      <div className={styles.apptFormSection}>
+        <label htmlFor="dob" className={styles.inputLabel}>
+          Date of birth:{' '}
+        </label>
+        <input
+          type="date"
+          id="dob"
+          className={styles.input}
+          {...register('dob', {
+            required: 'Date of birth is required.',
+          })}
+        />
+        <p className={styles.error}>{errors.dob && errors.dob.message}</p>
+      </div>
+      <div className={styles.apptFormSection}>
+        <label htmlFor="apptDate" className={styles.inputLabel}>
+          Appointment date:{' '}
+        </label>
+        <input
+          type="date"
+          id="apptDate"
+          className={styles.input}
+          {...register('apptDate', {
```

Comment:
Here, also we can discuss for future issue regarding - **Appointment** date selection is that not to allow people to select any date from the past and if we can take the holidays into account too, it will be great.

Reply:
Yes I figured that we can find a better calendar/date picker to use for this and the DOB category, but these are just  placeholders until we decide what we cant to use :)